libappstream-glib/as-image.c

```diff
@@ -830,8 +839,8 @@ as_image_save_pixbuf (AsImage *image,
 			      (gint) tmp_width,
 			      (gint) tmp_height,
 			      pixbuf,
-			      (gint) (width - tmp_width) / 2,
-			      (gint) (height - tmp_height) / 2);
+			      ABS ((gint) (width - tmp_width)) / 2,
+			      ABS ((gint) (height - tmp_height)) / 2);
```

Comment:
This ABS shouldn't be required, right? Since the above code should always ensure tmp_width <= width and tmp_height <= height. If you added this for safety you could put some assertions before this.

Reply:
@iainlane?

Reply:
sorry @hughsie @robert-ancell I forgot to reply. That also means I now forgot what the code does... let's see...

OK, here we go I think. I've added `if (pixbuf_width > width || pixbuf_height > height)`, which wasn't there before. In the `else` part of that `if` - the case where padding is required - I set `tmp_width` and `tmp_height` to the image's width and height.

That means that `width ≥ tmp_width` and `height ≥ tmp_height`, so `width - tmp_width` is always positive.

So in summary, I think the `ABS` is not needed. I'll rebase & remove it. :+1: